musica/micm/musica_ccpp_micm.F90

```diff
@@ -20,21 +20,27 @@ module musica_ccpp_micm
   !> Register MICM constituents with the CCPP
   subroutine micm_register(constituents, errmsg, errcode)
     use ccpp_constituent_prop_mod, only: ccpp_constituent_properties_t
-    use musica_util, only: error_t, mapping_t
-    type(ccpp_constituent_properties_t), allocatable, intent(out)   :: constituents(:)
-    character(len=512), intent(out) :: errmsg
-    integer, intent(out)            :: errcode
+    use musica_micm, only: Rosenbrock, RosenbrockStandardOrder
+    use musica_util, only: error_t
+
+    type(ccpp_constituent_properties_t), allocatable, intent(out) :: constituents(:)
+    character(len=512),                               intent(out) :: errmsg
+    integer,                                          intent(out) :: errcode
 
-    type(error_t) :: error
-    type(mapping_t) :: mapping
+    ! local variables
+    type(error_t)        :: error
     real(kind=kind_phys) :: molar_mass
-    logical :: is_advected
-    integer :: i
+    logical              :: is_advected
+    integer              :: solver_type
+    integer              :: num_grid_cells
+    integer              :: i
 
+    solver_type = Rosenbrock
+    num_grid_cells = 1
     errcode = 0
     errmsg = ''
 
-    micm => micm_t(filename_of_micm_configuration, error)
+    micm => micm_t(filename_of_micm_configuration, solver_type, num_grid_cells, error)
```

Comment:
Hi @mattldawson, do you think that `micm_register()` has `solver_type` and `num_grid_cells` as its function parameters or are we going set locally (kinda hard-coded as it is now)?